fvm/src/call_manager/mod.rs

```diff
@@ -42,7 +43,13 @@ pub trait CallManager: 'static {
     type Machine: Machine;
 
     /// Construct a new call manager.
-    fn new(machine: Self::Machine, gas_limit: i64, origin: (ActorID, Address), nonce: u64) -> Self;
+    fn new(
+        machine: Self::Machine,
+        gas_limit: i64,
+        origin: (ActorID, Address),
+        nonce: u64,
+        chain_context: ChainContext,
```

Comment:
The ChainContext is a Machine level field, not a CallManager level field. Let's hoist it up and remove it from here. (The ChainContext won't change across message applications within a single Machine instantiation)

Reply:
We can also move the `epoch` into the chain context, like @Stebalien suggested elsewhere.

Reply:
Ok, will hoist.

Reply:
done.